### PR TITLE
Remove surveys

### PR DESCRIPTION
**Pull request by yarneo**

### CODE REVIEW

File: src/main/java/com/mixpanel/android/mpmetrics/ConfigurationChecker.java [hunk 1]
```diff
@@ -176,6 +177,27 @@ public static boolean checkPushConfiguration(Context context) {
         return canRegisterWithPlayServices || canRegisterWithRegistrationIntent;
     }
 
+    public static boolean checkTakeoverInAppActivityAvailable(Context context) {
+        if (Build.VERSION.SDK_INT < MPConfig.UI_FEATURES_MIN_API) {
+            // No need to log, SurveyActivity doesn't work on this platform.
```
Comment (dmitry-novikov) [inline src/main/java/com/mixpanel/android/mpmetrics/ConfigurationChecker.java]: Still `SurveyActivity`?

File: src/main/java/com/mixpanel/android/mpmetrics/ConfigurationChecker.java [hunk 2]
```diff
@@ -176,6 +177,27 @@ public static boolean checkPushConfiguration(Context context) {
...
+            return false;
+        }
+
+        final Intent takeoverInAppIntent = new Intent(context, TakeoverInAppActivity.class);
+        takeoverInAppIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (dmitry-novikov) [inline src/main/java/com/mixpanel/android/mpmetrics/ConfigurationChecker.java]: Are this flags required to resolve activity in manifest?

File: src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java [hunk 3]
```diff
@@ -1823,7 +1823,7 @@ public void run() {
                         }
 
                         final InAppNotification.Type inAppType = toShow.getType();
-                        if (inAppType == InAppNotification.Type.TAKEOVER) {
+                        if (inAppType == InAppNotification.Type.TAKEOVER  && ! ConfigurationChecker.checkTakeoverInAppActivityAvailable(parent.getApplicationContext())) {
```
Comment (dmitry-novikov) [inline src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java]: Should we cache checker result? Operations with `PackageManager` are not lightweight and it's nonsense that registered manifest activities will be changed at run-time.

Comment (patedit) [inline src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java]: This is a great point @dmitry-novikov. Thanks!

**Pull request merged**